trailblazer/services/analysis_service/utils.py

```diff
@@ -5,31 +5,29 @@
 from trailblazer.store.models import Analysis
 
 
-def get_status_count(analyses: list[Analysis], status: TrailblazerStatus) -> int:
-    return len([a for a in analyses if a.status == status])
-
-
-def get_delivered_count(analyses: list[Analysis]) -> int:
-    return len([analysis for analysis in analyses if analysis.delivery])
+def get_status_counts(analyses: list[Analysis]) -> dict[TrailblazerStatus, int]:
+    """Returns the amount of analyses with each status."""
+    delivered: int = 0
+    status_counts: dict = {
+        TrailblazerStatus.CANCELLED: 0,
+        TrailblazerStatus.COMPLETED: 0,
+        TrailblazerStatus.FAILED: 0,
+        TrailblazerStatus.RUNNING: 0,
+    }
+    for analysis in analyses:
+        if analysis.delivery:
+            delivered += 1
+            continue
+        if analysis.status in status_counts.keys():
+            status_counts[analysis.status] += 1
```

Comment:
I guess this check is not technically necessary, since we will not render the amount of pending analyses for example. Maybe remove the check? @seallard

Reply:
I think it might be necessary due to how the dict is currently constructed. You could consider something like this as well
```python
from collections import defaultdict
from typing import List
from trailblazer.store.models import Analysis, TrailblazerStatus

def get_status_counts(analyses: List[Analysis]) -> dict:
    """Simplifies the counting of analysis statuses including delivered."""
    status_counts = defaultdict(int, {status: 0 for status in TrailblazerStatus})
    status_counts["delivered"] = 0

    for analysis in analyses:
        if analysis.delivery:
            status_counts["delivered"] += 1
        else:
            status_counts[analysis.status] += 1
    if "delivered" in status_counts:
        status_counts[TrailblazerStatus.COMPLETED] -= status_counts["delivered"] 
    return status_counts

def create_summary(analyses: List[Analysis], order_id: int) -> Summary:
    status_counts = get_status_counts(analyses)
    total = len(analyses)
    return Summary(order_id=order_id, total=total, **status_counts)
```

Reply:
That is probably cleaner, although the check for "delivered" in status_counts will be unnecessary with this way of counting 🥳

Reply:
Yeah, you are right - removed it 👍  Go for which ever implementation you think is better!

Reply:
Cleaned it up to something more akin to yours! Seems as if you do not need the defaultdict though, list comprehension support for dicts is a thing

Reply:
Nice, even better!